fileio/private/read_eeglabheader.m

```diff
@@ -38,6 +38,9 @@
 
 if ~isstruct(filename)
   load('-mat', filename, 'EEG');
+  if strcmp(lastwarn,'Variable ''EEG'' not found.') % EEGLAB > 2021.0 saves content of EEG as default
```

Comment:
Perhaps it would be better here not to rely on the warning, e.g. it could be that the user has non-standard settings for the warning behavior, in which case this might fail. Would it also work if you in advance check the contents of the file, using the output of a call to which.m, and then proceed depending on whether the variable 'EEG' exists or not?

Reply:
Do you have any specific solution in your mind?